internal/helper/helper.go

```diff
@@ -201,64 +232,94 @@ func CheckEmptyFields(Interface interface{}, result *gojsonschema.Result, schema
+			if err != nil {
+				return err
 			}
 		} else if field.Type().Kind() == reflect.Ptr {
 			// otherwise if it's just a pointer to a nested struct
 			// search it for empty required fields
-			if field.Elem().Kind() == reflect.Struct {
-				err := CheckEmptyFields(field.Interface(), result, schema)
-				if err != nil {
-					return err
-				}
+			err := checkEmptyFieldsInPtr(field, result, schema)
+			if err != nil {
+				return err
 			}
 		} else {
-
-			// check if the field is required and if it is present in the YAML file
-			required := false
-			tags := elem.Type().Field(i).Tag
-			jsonTag, hasJSON := tags.Lookup("json")
-			if hasJSON {
-				if !strings.Contains(jsonTag, "omitempty") {
-					required = true
-				}
+			tags, requiredFromTags := getTags(elem, i)
+			if !requiredFromTags && !isRequiredFromSchema(elem, i, schema) {
+				continue
 			}
-			// also check for required values in the jsonschema
-			for _, requiredField := range schema.Required {
-				if elem.Type().Field(i).Name == requiredField {
-					required = true
-				}
+			// this is a required field, check for zero values
+			if !reflect.Indirect(field).IsZero() {
+				continue
 			}
-			if required {
-				// this is a required field, check for zero values
-				if reflect.Indirect(field).IsZero() {
-					jsonContext := gojsonschema.NewJsonContext("image_definition", nil)
-					errDetail := gojsonschema.ErrorDetails{
-						"property": tags.Get("yaml"),
-						"parent":   elem.Type().Name(),
-					}
-					result.AddError(
-						newMissingFieldError(
-							gojsonschema.NewJsonContext("missing_field", jsonContext),
-							52,
-							errDetail,
-						),
-						errDetail,
-					)
-				}
+			jsonContext := gojsonschema.NewJsonContext("image_definition", nil)
+			errDetail := gojsonschema.ErrorDetails{
+				"property": tags.Get("yaml"),
+				"parent":   elem.Type().Name(),
 			}
+			result.AddError(
+				newMissingFieldError(
+					gojsonschema.NewJsonContext("missing_field", jsonContext),
+					52,
+					errDetail,
+				),
+				errDetail,
+			)
 		}
 	}
 	return nil
 }
 
+func checkEmptyFieldsInSlice(field reflect.Value, result *gojsonschema.Result, schema *jsonschema.Schema) error {
+	for i := 0; i < field.Cap(); i++ {
+		sliceElem := field.Index(i)
+		if sliceElem.Kind() == reflect.Ptr && sliceElem.Elem().Kind() == reflect.Struct {
+			err := CheckEmptyFields(sliceElem.Interface(), result, schema)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func checkEmptyFieldsInPtr(field reflect.Value, result *gojsonschema.Result, schema *jsonschema.Schema) error {
+	if field.Elem().Kind() == reflect.Struct {
+		err := CheckEmptyFields(field.Interface(), result, schema)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// getTags returns the json tags associated to the given element and checks if
+// these tags specify that the field is required.
+func getTags(elem reflect.Value, i int) (reflect.StructTag, bool) {
```

Comment:
hmm I dunno, for me `getTags()` shoud just... get the tags hehe. The other check should be another isRequiredSomething method imo.